log/klogv2/klogv2.go

```diff
@@ -31,68 +35,106 @@ var _ log.Logger = (*adapter)(nil)
 
 // adapter implements the `log.Logger` interface for klogv2
 type adapter struct {
-	fields map[string]interface{}
+	rawFields map[string]interface{}
+	fields    string
 }
 
-// FromKlogv2 creates a new `log.Logger` from the provided entry
-func FromKlogv2(fields map[string]interface{}) log.Logger {
-	return &adapter{fields}
+// New creates a new `log.Logger` from the provided entry
+func New(fields map[string]interface{}) log.Logger {
+	return &adapter{
+		rawFields: fields,
+		fields:    processFields(fields),
```

Comment:
I think now since we are calling this to append more fields, we can defer this to only process when writing to klog (and only process once per adapter). Maybe even wrap the map in a `Stringer`.

Reply:
I'll have to think about this with a fresh mind. Will come back to it in my morning time. Thanks!

Reply:
Done!